packages/patternfly-4/react-core/src/components/Nav/NavExpandable.js

```diff
@@ -44,28 +41,33 @@ const defaultProps = {
 
 class NavExpandable extends React.Component {
   id = this.props.id || getUniqueId();
+  state = {
+    expandedState: false
+  };
+  componentDidMount() {
+    this.setState({ expandedState: this.props.isExpanded });
+  }
+  componentDidUpdate(prevProps) {
+    if (this.props.isExpanded !== prevProps.isExpanded) {
```

Comment:
should this be:
```
  componentDidUpdate() {
    if (this.props.isExpanded !== this.state.isExpanded) {
```
?

Reply:
sorry:
```
if (this.props.isExpanded !== prevProps.isExpanded || this.props.isExpanded !== this.state.expandedState)
```

Reply:
ok, made a few changes to address this. If the consumer is setting expandState externally, we should allow them to hook into this, otherwise they could be become out of sync. My changes here would allow the consumer to override the expand/collapse state at either the top level (Nav and the `onToggle`), or at a more granular level, the `NavExpandable` and the `onExpand` callback). They would then be able to hook into either (or neither if desired). See if this works on your side...
https://github.com/priley86/patternfly-react/commit/ee30e2002615f561c93f5ddf471678fd423f1b9a

Reply:
downstream explanation/relevant changes...
https://github.com/openshift/console/pull/1238#issuecomment-469763337